exporters/otlp/grpcsingle.go

```diff
@@ -0,0 +1,91 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlp // import "go.opentelemetry.io/otel/exporters/otlp"
+
+import (
+	"context"
+	"sync"
+
+	"google.golang.org/grpc"
+
+	colmetricpb "go.opentelemetry.io/otel/exporters/otlp/internal/opentelemetry-proto-gen/collector/metrics/v1"
+	coltracepb "go.opentelemetry.io/otel/exporters/otlp/internal/opentelemetry-proto-gen/collector/trace/v1"
+	metricsdk "go.opentelemetry.io/otel/sdk/export/metric"
+	tracesdk "go.opentelemetry.io/otel/sdk/export/trace"
+)
+
+type grpcSingleConnectionManager struct {
+	connection *grpcConnection
+
+	lock          sync.Mutex
+	metricsClient colmetricpb.MetricsServiceClient
+	tracesClient  coltracepb.TraceServiceClient
+}
+
+func (m *grpcSingleConnectionManager) getMetricsClient() grpcMetricsClient {
+	return grpcMetricsClient{
+		grpcClientBase: grpcClientBase{
+			clientLock: &m.lock,
+			connection: m.connection,
+		},
+		client: m.metricsClient,
+	}
+}
+
+func (m *grpcSingleConnectionManager) getTracesClient() grpcTracesClient {
+	return grpcTracesClient{
+		grpcClientBase: grpcClientBase{
+			clientLock: &m.lock,
+			connection: m.connection,
+		},
+		client: m.tracesClient,
+	}
+}
+
+func (m *grpcSingleConnectionManager) handleNewConnection(cc *grpc.ClientConn) error {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	if cc != nil {
+		m.metricsClient = colmetricpb.NewMetricsServiceClient(cc)
+		m.tracesClient = coltracepb.NewTraceServiceClient(cc)
+	} else {
+		m.metricsClient = nil
+		m.tracesClient = nil
+	}
+	return nil
+}
+
+func NewGRPCSingleConnectionManager(cfg GRPCConnectionConfig) ConnectionManager {
```

Comment:
`NewGRPCSingleConnectionManager` implies there is something other than "Single" that we would want to add in the future.  Would we want to make the "other" thing we want to add an option instead of a new function, and shorten this to `NewGRPCConnectionManager`?  That would be fine so long as the options would still apply, which I think they should.